src/aatoolbox/datasources/iri/iri_forecast_seasonal_precipitation_tercile.py

```diff
@@ -0,0 +1,188 @@
+            cookies={"__dlauth_id": iri_auth},
+        )
+        with open(output_filepath, "wb") as out_file:
+            out_file.write(response.content)
+
+        return output_filepath
+
+    def _get_url(self, dominant: bool):
+        # question: best to define url here or as global constant?
+        base_url = (
+            "https://iridl.ldeo.columbia.edu/SOURCES/.IRI/.FD/"
+            ".NMME_Seasonal_Forecast/.Precipitation_ELR/"
+        )
+        if dominant:
+            base_url += ".dominant/"
+        else:
+            base_url += ".prob/"
+        return (
+            f"{base_url}"
+            f"X/%28{self._geobb.west}%29%28{self._geobb.east}%29RANGEEDGES/"
+            f"Y/%28{self._geobb.north}%29%28{self._geobb.south}%29RANGEEDGES/"
+            "data.nc"
+        )
+
+    def load(self, dominant: bool = False):
+        """Preprocess and load the IRI forecast.
+
+        dominant: bool
+            Two datasets with the tercile forecast exists. One reports
+            the probability per tercile, which is loaded with
+            dominant=False. The other only reports the probability
+            of the dominant tercile, which is loaded with
+            dominant=True. In this data low negative probabilities indicate the
+            below average tercile is dominant and high positive that the above
+            average tercile is dominant.
+
+        Examples
+        --------
+        >>> from aatoolbox.pipeline import Pipeline
+        >>> from aatoolbox.utils.geoboundingbox import GeoBoundingBox
+        >>> (from aatoolbox.datasources.iri.
+        ... iri_forecast_seasonal_precipitation_tercile import IriForecast)
+        >>> iso3="bfa"
+        >>> pipeline_iso = Pipeline(iso3)
+        >>> codab_admin1 = pipeline_iso.load_codab(admin_level=1)
+        >>> geo_bounding_box = GeoBoundingBox.from_shape(codab_admin1)
+        >>> iri=IriForecast(iso3,geo_bounding_box)
+        >>> iri.load()
+        """
+        ds = xr.load_dataset(
+            self._get_raw_path(dominant),
+            decode_times=False,
+            drop_variables="C",
+        )
+        # fix dates
+        ds.aat.set_time_dim(t_dim="F", inplace=True)
+        ds.aat.correct_calendar(inplace=True)
+        ds = xr.decode_cf(ds)
+
+        # question: Do you think this makes more sense to do this here
+        # or implement in the geoboundingbox class? i.e. restrict
+        # the range and that south<north?
+        # IRI downloads in the order you give the coordinates
+        # and accepts both -180 to 180 longitudes and 0 to 360
+        ds.aat.invert_coordinates(inplace=True)
+        ds.aat.change_longitude_range(inplace=True)
+
+        # question: I now chose to rename the dims to imo more logical names
+        # however, this does mean it is not the original data anymore
+        # and we will have to change it in all analyses
+        # do you think it makes sense to rename?
```

Comment:
Just had a quick look cause I got done adding tests to the raster class. Here, you don't need to rename anything. The raster module builds on the automatic x and y dim detection in rioxarray. These are accessible on any dataset or dataarray using `ds.aat.x_dim` and `ds.aat.y_dim`, in the same way that `t_dim` is accessible. Thus any analyses can just rely on these class attributes, and only for datasets that don't have one of the defaults would you have to manually set these values, using `ds.aat.x_dim = "a"` for instance. This is based on the rioxarray methods. The defaults should cover most cases, though, which are `X, x, lon, longitude` and corresponding values for `y`.

So, since you've already set `t_dim = "F"`, you can now just use `ds.aat.x_dim`, `ds.aat.y_dim`, and `ds.aat.t_dim` if access is ever required and we can maintain dimension naming similar to the original source.

Reply:
Ohh awesome, thanks, missed that! 

Only thing would be that it is maybe less clear to users what the dimensions (e.g. F) mean, and you would probably still often use the names, e.g. ds.sel(`F="2021-01-16"`)  But I think that doesn't outweight keeping the data as is, and we can maybe document the meaning of the dimensions in the code